pkg/policies/policies.go

```diff
@@ -204,3 +170,35 @@ func policyViolationsToAttestationViolations(violations []*engine.PolicyViolatio
 	}
 	return
 }
+
+// We need to create a connection for every single artifact, because it depends on the digest
+func (pv *PolicyVerifier) getCASClient(role pb.CASCredentialsServiceGetRequest_Role, digest string) (*casclient.Client, error) {
```

Comment:
I believe we have some code that does this already, so we could just pass the connection ready to the action?

Reply:
https://github.com/migmartri/chainloop/blob/40f96e6edc5c4f3bba347b30b44af88614dd8e30/app/cli/cmd/artifact.go#L38-L62

Reply:
It needs the digest, which depends on each material. We might want to revisit that restriction, but currently we must create a new client for each.